File: java/basic_app/app/src/main/java/com/appsflyer/onelink/appsflyeronelinkbasicapp/applesActivity.java
package com.appsflyer.onelink.appsflyeronelinkbasicapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ListView;
import android.widget.TextView;

import com.appsflyer.deeplink.DeepLink;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

import static com.appsflyer.onelink.appsflyeronelinkbasicapp.AppsflyerBasicApp.LOG_TAG;

public class applesActivity extends AppCompatActivity {
    Utilities utilities;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_apples);

        Intent intent = getIntent();
        Gson json = new Gson();
        DeepLink dlDataJ = json.fromJson(intent.getStringExtra(AppsflyerBasicApp.DL_ATTRS), DeepLink.class);
        Map dlData = json.fromJson(String.valueOf(dlDataJ), Map.class);

        if (dlData != null) {
            MapToListViewAdapter adapter = new MapToListViewAdapter(dlData);
            ListView dlAttrsList = findViewById(R.id.apples_deeplinkparamslist);
            dlAttrsList.setAdapter(adapter);
        } else {
            TextView dlTitleText = findViewById(R.id.apples_deeplinktitle);
            dlTitleText.setText("No Deep Linking Happened");
        }

    }

    public void goToConversionData(View view) {
        utilities = new Utilities();
        //TODO - move redirection to the Utilities class
        Utilities.goToConversionData(view);
        //TODO - add log actions
        Intent intent = new Intent(getApplicationContext(), conversionDataActivity.class);
        startActivity(intent);
    }


}

File: kotlin/basic_app/app/src/main/kotlin/com/appsflyer/onelink/appsflyeronelinkbasicapp/ApplesActivity.kt
package com.appsflyer.onelink.appsflyeronelinkbasicapp

import android.os.Bundle
import android.util.Log
import com.appsflyer.AppsFlyerLib
import com.appsflyer.AFInAppEventType // Predefined event names
import com.appsflyer.AFInAppEventParameterName // Predefined parameter names
import com.appsflyer.attribution.AppsFlyerRequestListener
import com.appsflyer.onelink.appsflyeronelinkbasicapp.AppsflyerBasicApp.Companion.LOG_TAG

class ApplesActivity : FruitActivity() {
    override val fruitName: String = "apples"
    override fun getLayoutResourceId(): Int = R.layout.activity_apples

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val eventValues = HashMap<String, Any>()
        eventValues[AFInAppEventParameterName.PRICE] = 1234.56
        eventValues[AFInAppEventParameterName.CONTENT_ID] = "1234567"

        AppsFlyerLib.getInstance().logEvent(
            applicationContext,
            AFInAppEventType.ADD_TO_WISH_LIST , eventValues)

        AppsFlyerLib.getInstance().logEvent(
            applicationContext,
            AFInAppEventType.PURCHASE,
            eventValues,
            object : AppsFlyerRequestListener {
                override fun onSuccess() {
                    Log.d(LOG_TAG, "Event sent successfully")
                }
                override fun onError(errorCode: Int, errorDesc: String) {
                    Log.d(LOG_TAG, "Event failed to be sent:\n" +
                            "Error code: " + errorCode + "\n"
                            + "Error description: " + errorDesc)
                }
            })
    }

}